zilliqa/src/pool.rs

```diff
@@ -289,14 +293,41 @@ impl TransactionPool {
             Self::add_to_gas_index(&mut self.gas_index, &txn);
         }
 
+        // If this is a transaction created at this node, add it to broadcast vector
+        if !from_broadcast {
+            self.store_broadcast_txn(txn.tx.clone());
+        }
+
         debug!("Txn added to mempool. Hash: {:?}, from: {:?}, nonce: {:?}, account nonce: {account_nonce}", txn.hash, txn.signer, txn.tx.nonce());
 
         // Finally we insert it into the tx store and the hash reverse-index
         self.hash_to_index.insert(txn.hash, txn.mempool_index());
         self.transactions.insert(txn.mempool_index(), txn);
+
         TxAddResult::AddedToMempool
     }
 
+    fn store_broadcast_txn(&mut self, txn: SignedTransaction) {
+        self.transactions_to_broadcast.push_back(txn);
+    }
+
+    pub fn pull_txns_to_broadcast(&mut self) -> Result<Vec<SignedTransaction>> {
+        const MAX_BATCH_SIZE: usize = 100;
```

Comment:
Might want to make this 400, which would be a theoretical max of 400/100ms (4K/s)